actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +364,14 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.class.in? [ActiveStorage::Variant, ActiveStorage::Blob, ActiveStorage::Attachment]
```

Comment:
Wondering if we can't do it the other way around. If the source is a string/symbol or whatever was reasonable before to use with image_tag, we'll call the existing path_to_image. If it's not, we try polymorphic_url. Then we wouldn't have to hardcode this to Active Storage and would open the door for people to build their own direct URLs for images.